src/PrayerTimes.js

```diff
@@ -98,14 +110,34 @@ export default class PrayerTimes {
         var maghribAdjustment = (calculationParameters.adjustments.maghrib || 0) + (calculationParameters.methodAdjustments.maghrib || 0);
         var ishaAdjustment = (calculationParameters.adjustments.isha || 0) + (calculationParameters.methodAdjustments.isha || 0);
 
-        this.fajr = roundedMinute(dateByAddingMinutes(fajrTime, fajrAdjustment));
-        this.sunrise = roundedMinute(dateByAddingMinutes(sunriseTime, sunriseAdjustment));
+        this.fajr = this.safePolarNightMidnightSun(roundedMinute(dateByAddingMinutes(fajrTime, fajrAdjustment)), 'fajr');
```

Comment:
I don't think we should adjust each prayer time individually, if an adjustment is needed then the prayer times should be consistent with that adjustment. l would recommend at the beginning of this function checking if sunrise or sunset is invalid and then adjusting params there.

Reply:
I will reply to that in your comment below